githubbot/githubbot/handler.go

```diff
@@ -109,160 +104,139 @@ func (h *Handler) handleSubscribe(cmd string, msg chat1.MsgSummary, create bool,
 
 	args := toks[2:]
 	if len(args) < 1 {
-		var message string
 		if create {
-			message = "I don't understand! Try `!github subscribe <username/repo>`"
+			h.ChatEcho(msg.ConvID, "I don't understand! Try `!github subscribe <username/repo>`")
 		} else {
-			message = "I don't understand! Try `!github unsubscribe <username/repo>`"
-		}
-		_, err = h.kbc.SendMessageByConvID(msg.ConvID, message)
-		if err != nil {
-			err = fmt.Errorf("error sending message: %s", err)
+			h.ChatEcho(msg.ConvID, "I don't understand! Try `!github unsubscribe <username/repo>`")
 		}
-		return err
+		return nil
 	}
 
+	repo := args[0]
 	// Check if command is subscribing to a branch
+	alreadyExists, err := h.db.GetSubscriptionForRepoExists(msg.ConvID, repo)
+	if err != nil {
+		return fmt.Errorf("error checking subscription: %s", err)
+	}
 	if len(args) == 2 {
+		if !alreadyExists {
+			if create {
+				if err := h.handleNewSubscription(repo, msg, client); err != nil {
+					return err
+				}
+			} else {
+				h.ChatEcho("You aren't subscribed to notifications for `%s`!", repo)
+				return nil
+			}
+		}
 		switch args[1] {
 		case "issues", "pulls", "statuses", "commits":
-			return h.handleSubscribeToFeature(args[0], args[1], msg, create)
+			return h.handleSubscribeToFeature(repo, args[1], msg, create)
```

Comment:
you probably need to run `setFeatures` with all of the features set to false before this, otherwise you're just running `subscribeToFeature` with the default prefs (everything true)

Reply:
oops i misread that we were doing a get first, will fix